src/main/js/controls/Pie/PieConfig.js

```diff
@@ -1,9 +1,51 @@
 "use strict";
 import BaseConfig, { getDefaultValue } from "../../core/BaseConfig";
 import { generateClipPathId } from "../../core/BaseConfig/helper";
+import constants from "../../helpers/constants";
 import errors from "../../helpers/errors";
 import utils from "../../helpers/utils";
 
+/**
+ * Helper function to set the right legend padding values based on input JSON.
+ *
+ * @private
+ * @param {object} config - config which needs to be updated
+ * @param {object} inputLegendPadding - input legend padding provided via input JSON.
+ * @returns {object} - padding for Legend
+ */
+const getLegendPadding = (config, inputLegendPadding) => {
+    // If legendPadding is provided, update the config object with the provided values, else update it with the default constants
+    if (utils.isDefined(config.legendPadding)) {
+        return {
+            top: getDefaultValue(
+                inputLegendPadding.top,
+                constants.PIE_LEGEND_PADDING.top
+            ),
+            bottom: getDefaultValue(
+                inputLegendPadding.bottom,
+                constants.PIE_LEGEND_PADDING.bottom
+            ),
+            right: getDefaultValue(
+                inputLegendPadding.right,
+                constants.PIE_LEGEND_PADDING.right
+            ),
+            left: getDefaultValue(
+                inputLegendPadding.left,
+                constants.PIE_LEGEND_PADDING.left
+            ),
+            hasCustomLegendPadding: true
+        };
+    } else {
+        return {
+            top: constants.PIE_LEGEND_PADDING.top,
+            bottom: constants.PIE_LEGEND_PADDING.bottom,
+            right: constants.PIE_LEGEND_PADDING.right,
+            left: constants.PIE_LEGEND_PADDING.left,
+            hasCustomLegendPadding: false
+        };
+    }
+};
```

Comment:
It looks like the same method and implementation using multiple files. Can we move this function helper function and use the same helper function in the files?

Reply:
Moved this method to helpers/legend.js [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/046e78e57ec9658a1d0ea0ba47837e4cf6772dbd).